client/src/pages/CodeUI.js

```diff
@@ -51,15 +54,27 @@ function CodeUI(props) {
   const [question, setQuestion] = useState(" ");
   const [interview, setInterview] = useState(interviewTitle);
   const [runResult, setrunResult] = useState(null);
+  const [socket, setSocket] = useState();
+
   useEffect(() => {
     setQuestion(premadeq);
-  });
+    let socket = socketIOClient(ENDPOINT);
```

Comment:
be careful with const and let - use const whenever you can

Reply:
Shouldn't you be importing socket from the socket/socket file - so you initialize it only once?

Reply:
This was for the demo